### ISSUE TITLE
Latexify fails to properly format statements containing arrays

### ISSUE THREAD

**Issue by dmeklund** [korsbo/Latexify.jl#85]
- Latexify version:  0.12.2
- Julia version: 1.2.0

### Summary
 While expressions containing arrays are generally formatted correctly by Latexify, statements such as `x=[1 0]` either fail to properly format the array contents, or (in the case of array literals) result in errors.

#### Arrays in statements
```
julia> latexify(zeros(2))
L"\begin{equation}
\left[
\begin{array}{c}
0.0 \\
0.0 \\
\end{array}
\right]
\end{equation}
"

julia> latexify(:(x=$(zeros(2))))
L"$x = [0.0, 0.0]$"
```
Expected output: the same LaTeX as in the `latexify(zeros(2))` call, but prepended with "x =".

More complex arrays can lead to illegible output:

```
julia> arr=[0 -im; im 0]
2×2 Array{Complex{Int64},2}:
 0+0im  0-1im
 0+1im  0+0im

julia> latexify(:(x=$(arr)))
L"$x = Complex{Int64}[0 + 0im 0 - 1im; 0 + 1im 0 + 0im]$"
```

#### Array literals

Using array literals as an input to latexify results in errors for both statements and expressions. This is more of an issue for statements where literals are more likely to be convenient (e.g. in the case above where I had to explicitly assign `arr` first in order to avoid using a literal):

```
julia> latexify([0 1])
L"\begin{equation}
\left[
\begin{array}{cc}
0 & 1 \\
\end{array}
\right]
\end{equation}
"                                                                                                                                                                 

julia> latexify(:[0 1])
ERROR: Latexify.jl's latexoperation does not know what to do with one of the
                operators in your expression (0).
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] #latexoperation#19(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Latexify.latexoperation), ::Expr, ::Array{Symbol,1
}) at /home/david/.julia/packages/Latexify/0gnwF/src/latexoperation.jl:127
 [3] latexoperation at /home/david/.julia/packages/Latexify/0gnwF/src/latexoperation.jl:10 [inlined]
 [4] (::getfield(Latexify, Symbol("#recurseexp!#4")){Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}})(::Expr) at /home/david/.julia/packages/
Latexify/0gnwF/src/latexraw.jl:71
 [5] #latexraw#3(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(latexraw), ::Expr) at /home/david/.julia/packages/Latexif
y/0gnwF/src/latexraw.jl:74
 [6] latexraw at /home/david/.julia/packages/Latexify/0gnwF/src/latexraw.jl:62 [inlined]
 [7] #latexinline#42 at /home/david/.julia/packages/Latexify/0gnwF/src/latexinline.jl:2 [inlined]
 [8] latexinline(::Expr) at /home/david/.julia/packages/Latexify/0gnwF/src/latexinline.jl:2
 [9] #latexify#25(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(latexify), ::Expr) at /home/david/.julia/packages/Latexify/0gnwF
/src/latexify_function.jl:11
 [10] latexify(::Expr) at /home/david/.julia/packages/Latexify/0gnwF/src/latexify_function.jl:4
 [11] top-level scope at REPL[42]:1

julia> latexify(:(x=[0 1]))
ERROR: MethodError: Cannot `convert` an object of type Int64 to an object of type Symbol
Closest candidates are:
  convert(::Type{T}, ::T) where T at essentials.jl:167
  Symbol(::Any...) at strings/basic.jl:206
Stacktrace:
 [1] setindex!(::Array{Symbol,1}, ::Int64, ::Int64) at ./array.jl:766
 [2] (::getfield(Latexify, Symbol("#recurseexp!#4")){Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}})(::Expr) at /home/david/.julia/packages/
Latexify/0gnwF/src/latexraw.jl:67
 [3] #latexraw#3(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(latexraw), ::Expr) at /home/david/.julia/packages/Latexif
y/0gnwF/src/latexraw.jl:74
 [4] latexraw at /home/david/.julia/packages/Latexify/0gnwF/src/latexraw.jl:62 [inlined]
 [5] #latexinline#42 at /home/david/.julia/packages/Latexify/0gnwF/src/latexinline.jl:2 [inlined]
 [6] latexinline(::Expr) at /home/david/.julia/packages/Latexify/0gnwF/src/latexinline.jl:2
 [7] #latexify#25(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(latexify), ::Expr) at /home/david/.julia/packages/Latexify/0gnwF
/src/latexify_function.jl:11
 [8] latexify(::Expr) at /home/david/.julia/packages/Latexify/0gnwF/src/latexify_function.jl:4
 [9] top-level scope at REPL[43]:1
```

### Workaround

Generally you can get something that displays fine by simply appending the statement portion of the LaTeX explicitly using LaTeXStrings (being careful to avoid literals), but the output is technically wrong since the statement portion lies outside of the equation block.

```
julia> arr=[0 -im; im 0]
2×2 Array{Complex{Int64},2}:
 0+0im  0-1im
 0+1im  0+0im

julia> latexstring("x="*latexify(arr).s)
L"$x=\begin{equation}
\left[
\begin{array}{cc}
0+0\textit{i} & 0-1\textit{i} \\
0+1\textit{i} & 0+0\textit{i} \\
\end{array}
\right]
\end{equation}
$"
```

### For reference

```
julia> dump(:(x=$(zeros(2))))
Expr
  head: Symbol =
  args: Array{Any}((2,))
    1: Symbol x
    2: Array{Float64}((2,)) [0.0, 0.0]

julia> dump(:(x=[0, 2]))
Expr
  head: Symbol =
  args: Array{Any}((2,))
    1: Symbol x
    2: Expr
      head: Symbol vect
      args: Array{Any}((2,))
        1: Int64 0
        2: Int64 2

julia> dump(:(x=[0 2]))
Expr
  head: Symbol =
  args: Array{Any}((2,))
    1: Symbol x
    2: Expr
      head: Symbol hcat
      args: Array{Any}((2,))
        1: Int64 0
        2: Int64 2

julia> dump(:(x=[0;2]))
Expr
  head: Symbol =
  args: Array{Any}((2,))
    1: Symbol x
    2: Expr
      head: Symbol vcat
      args: Array{Any}((2,))
        1: Int64 0
        2: Int64 2
```

**Comment by korsbo** [comment]
Thanks for your thorough report! It should be fixed by #86 but there might still be some kinks. Feel free to open further issues if you encounter some of them.

**Comment by korsbo** [comment]
Released as `v0.12.3`
